tests/test_bot.py

```diff
@@ -641,3 +641,16 @@ def post(*args, **kwargs):
 
         with pytest.raises(OkException):
             bot.send_photo(chat_id, open('tests/data/telegram.jpg', 'rb'), timeout=timeout)
+
+    def test_timeout_propagation_implicit(self, monkeypatch, bot, chat_id):
+        class OkException(Exception):
```

Comment:
Interesting way to test this (with the exception). I know it was already in, but why don;t we just assert in the monkeypatched method? That should work.